internal/request/request_test.go

```diff
@@ -34,7 +34,6 @@ func (task *fakeCatalogTask) Get() (*common.RequestMessage, error) {
 		Input: common.RequestInput{
 			ResponseFormat: "tar",
 			Jobs: []common.JobParam{
-				{Method: "monitor", HrefSlug: "/api/v2/jobs/7008"},
```

Comment:
@mkanoor How can this fix the problem? In reality the Jobs can contain multiple jobs.

Reply:
@bzwei The problem is in the fakeHandler which is using the counter variable. When called multiple times by different go routines they are incrementing the same variable. If you look at our workHandler it only has methods it doesn't have any variables to update like in the fake Handler. So if you want to monitor if 2 requests were processed that variable needs to be at a higher level and not in the worker. Remember we had talked about a counter that tracks the number of requests handled so far.